airbrakes/hardware/imu.py

```diff
@@ -123,36 +203,93 @@ def _fetch_data_loop(self, port: str, frequency: int, _: bool):
                 # Put the latest data into the shared queue
                 self._data_queue.put(imu_data_packet)
 
-    def get_imu_data_packet(self) -> IMUDataPacket:
-        """
-        Gets the last available data packet from the IMU.
 
-        Note: If `get_imu_data_packet` is called slower than the frequency set, the data will not
-        be the latest, but the first in the queue.
+class MockIMU(AbstractIMU):
```

Comment:
I don't see why you have to make an Abstract class just to create a `MockIMU`? Moreover this does not belong in `airbrakes/`, it should either be in `sims/` or `tests/`. 

One can just create a subclass and override any method you want, it is not necessary to create ABC's in python. 

ABC's can make sense if we have many different versions of the `IMU` class, but having a `Test/MockIMU` should not necessiate the change.

Reply:
I want both IMU and MockIMU to have the same superclass, so how would you recommend having the file structure if we did it that way? Putting it all in the same file was the cleanest way I could think but idk

Reply:
keep the IMU class as is. `MockIMU` can go in `sims/`, and just inherit from `IMU`. You can override any of the methods you want, e.g.

```python
class MockIMU(IMU):
    def _fetch_data_loop(self, *args, **kwargs):
        # logic for reading file and putting packets in queue here
```

From what I can see, the only change you want in the MockIMU class is to read data from the csv file as packets. This is similar to what I did in the tests, see e.g. `test_data_packets_fetch` in `test_imu.py`. In that test, I didn't even need to subclass. Depending on how much you want to change, subclassing may be a better option. However, the main files inside `airbrakes` must remain unchanged (unless you see a bug or something ofc)

Reply:
Okay that sounds good